windows/src/engine/keyman32/kmprocessactions.cpp

```diff
@@ -7,21 +7,33 @@
 #include "pch.h"
 
 /**
-* Adds characters to the queue to be inserted into the application
+* Adds converts the characters to UTF-16 and inserts CR as it places them
+* in the queue. 
 * @param app           A pointer to the AITIP instance.
-* @param outputString  A null-terminated string of characters to insert into the application
+* @param core_output   A null-terminated string of characters in UTF-32 format to insert into the application
 */
 
+// When the windows platform code is updated to not use QueueAction
+// This function would be better to take as input the km_core_usv*
+// string form the core and generate a new buffer that is
+// windows formated (CRLF) and encoded as UTF-16. Currently it just
+// does both in one parse while adding it to the queue to reduce the double
+// processing.
+
 static void
-processUnicodeChar(AITIP* app, const km_core_usv* outputString) {
-  while (*outputString) {
-    if (Uni_IsSMP(*outputString)) {
-      app->QueueAction(QIT_CHAR, (Uni_UTF32ToSurrogate1(*outputString)));
-      app->QueueAction(QIT_CHAR, (Uni_UTF32ToSurrogate2(*outputString)));
+process_output_string(AITIP* app, const km_core_usv* core_output) {
+  while (*core_output) {
+    if (*core_output == L'\n') {
+      // Insert '\r' for Windows platform applications
+      app->QueueAction(QIT_CHAR, L'\r');
+    }
```

Comment:
It makes no practical difference, I know, but for clarity can we put this test inside the `else` clause below?